src/routing/ProtectedRoute.tsx

```diff
@@ -21,7 +24,28 @@ export const ProtectedRoute = ({
   component: WrappedComponent,
   ...rest
 }: RouteProps): JSX.Element => {
-  const { displayedName, isLoading } = useLoginContext()
+  const {
+    displayedName,
+    isLoading,
+    userId,
+    userType,
+    email,
+    contactNumber,
+  } = useLoginContext()
+  const growthbook = useGrowthBook()
+  const currPath = useLocation().pathname
+
+  useEffect(() => {
+    if (growthbook)
+      growthbook.setAttributes({
+        userId,
+        userType,
+        email,
+        displayedName,
+        contactNumber,
+        siteName: getSiteNameAttributeFromPath(currPath),
```

Comment:
can we conditionally set the siteName iff it exists? idk how growthbook works but idw us to have a phantom site that's essentially "every page without a site"